lib/iris/util.py

```diff
@@ -1554,3 +1554,73 @@ def _meshgrid(*xi, **kwargs):
         if mxii.dtype != xii.dtype:
             mxi[i] = mxii.astype(xii.dtype)
     return mxi
+
+
+def find_discontiguities(coord, abs_tol=1e-4):
+    """
+    Searches coord for discontiguities in the bounds array, returned as
+    booleans.
+
+    Args:
+        coord:
+        An instance of :class:`iris.coords.Coord`
+
+    Kwargs:
+        abs_tol:
+        absolute tolerance in degrees
+
+    Returns:
+        Boolean array of values representing the truth value for
+        discontiguous cells in the coordinate points array.
+    """
+    _, diffs_x, diffs_y = iris.coords._discontiguity_in_2d_bounds(coord.bounds,
+                                                                  abs_tol)
+
+    gaps_x = diffs_x > abs_tol
+    gaps_y = diffs_y > abs_tol
+
+    # Set up unmasked boolean array the same size as the coord points array:
+    bad_points_boolean = np.zeros((coord.points.shape), dtype=bool)
+
+    # Apply mask for x-direction discontiguities:
+    bad_points_boolean[:, 1:] = np.logical_or(bad_points_boolean[:, 1:],
+                                              gaps_x)
+
+    # apply mask for y-direction discontiguities:
+    bad_points_boolean[1:, :] = np.logical_or(bad_points_boolean[1:, :],
+                                              gaps_y)
+
+    try:
+        return bad_points_boolean
+    except ValueError:
+        return None
+
+
+def mask_discontiguities(coord, data, abs_tol=1e-4):
+    """
+    Masks any cells in the data array which correspond to discontiguous
+    cells in the coordinate points array.
+
+    Args:
+        coord:
+        An instance of :class:`iris.coords.Coord`
+
+        data:
+        the :attr:`data` of the cube from which the coord is taken.
+
+    Kwargs:
+        abs_tol:
+        absolute tolerance in degrees
+
+    Returns:
+        Copy of the data array, masked at points where bounds are
+        discontiguous.
+    """
+    mask = find_discontiguities(coord, abs_tol=abs_tol)
```

Comment:
`mask` isn't that clear what it means as it doesn't refer to the cube's original mask, it refers to the locations of the discontiguities and so the new places that should be masked.
How about something like `discontiguous_locations` or `discontiguous` or `discontiguous_bounds_locations`... I can't really think of a good name.

Reply:
Maybe just `new_mask` or `extra_mask`